core/src/main/java/io/temporal/samples/taskinteraction/WorkflowWithTasksImpl.java

```diff
@@ -0,0 +1,76 @@
+/*
+ *  Copyright (c) 2020 Temporal Technologies, Inc. All Rights Reserved
+ *
+ *  Copyright 2012-2016 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ *  Modifications copyright (C) 2017 Uber Technologies, Inc.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License"). You may not
+ *  use this file except in compliance with the License. A copy of the License is
+ *  located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ *  or in the "license" file accompanying this file. This file is distributed on
+ *  an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ *  express or implied. See the License for the specific language governing
+ *  permissions and limitations under the License.
+ */
+
+package io.temporal.samples.taskinteraction;
+
+import io.temporal.workflow.Async;
+import io.temporal.workflow.Promise;
+import io.temporal.workflow.Workflow;
+import java.util.Arrays;
+import org.slf4j.Logger;
+
+/** Workflow that creates three task and waits for them to complete */
+public class WorkflowWithTasksImpl implements WorkflowWithTasks {
+
+  private final Logger logger = Workflow.getLogger(WorkflowWithTasksImpl.class);
+
+  private final TaskService<Void> taskService = new TaskService<>();
+
+  @Override
+  public void execute() {
+
+    // Schedule two "tasks" in parallel. The last parameter is the token the client needs
+    // to unblock/complete the task. This token contains the workflowId that the
+    // client can use to create the workflow stub.
+    final TaskToken taskToken = new TaskToken();
+
+    logger.info("About to create async tasks");
+    final Promise<Void> task1 =
+        Async.procedure(
+            () -> {
+              final Task task = new Task(taskToken.getNext(), new Task.TaskTitle("TODO 1"));
+              taskService.executeTask(task);
+            });
+
+    final Promise<Void> task2 =
+        Async.procedure(
+            () -> {
+              final Task task = new Task(taskToken.getNext(), new Task.TaskTitle("TODO 2"));
+              taskService.executeTask(task);
+            });
+
+    logger.info("Awaiting for the two tasks to complete");
+    // Block execution until both tasks complete
+    Promise.allOf(Arrays.asList(task1, task2)).get();
```

Comment:
i think here you need to loop through promises again to check if any failed and if not call .get on each (to make sure they both actually compelted)

Reply:
I get your point and i can add it, but I don't think any of these promises can fail with the current implementation 

https://github.com/temporalio/samples-java/pull/567/files#diff-89e67e1da37a2cdfafecf424ebbae1cd65645bdf244e414e802dafd2fdf5208aR88